nevergrad/functions/helpers.py

```diff
@@ -0,0 +1,123 @@
+            setattr(instance, name, getattr(self, name).copy())
+        return instance
+
+    def compute_pseudotime(  # pylint: disable=unused-argument
+        self, input_parameter: tp.Any, loss: tp.Loss
+    ) -> float:
+        return self._experiment.compute_pseudotime(input_parameter, loss)
+
+    def evaluation_function(self, *recommendations: p.Parameter) -> float:
+        if self._pareto_size is not None and len(recommendations) > self._pareto_size:
+            # select a subset
+            hypervolume = mobj.HypervolumePareto(upper_bounds=self.multiobjective_upper_bounds)
+            hypervolume.extend(recommendations)
+            recommendations = tuple(
+                hypervolume.pareto_front(
+                    size=self._pareto_size,
+                    subset=self._pareto_subset,
+                    subset_tentatives=self._pareto_subset_tentatives,
+                )
+            )
+        return min(self._evaluation.evaluation_function(recom) for recom in recommendations)
+
+    @property
+    def descriptors(self) -> tp.Dict[str, tp.Any]:
+        """Description of the function parameterization, as a dict. This base class implementation provides function_class,
+        noise_level, transform and dimension
+        """
+        desc = dict(self._descriptors)
+        desc.update(self._experiment.descriptors)
+        # TODO descriptors for the evaluation function?
+        return desc
+
+    @classmethod
+    def create_crossvalidation_experiments(
+        cls,
+        experiments: tp.List[base.ExperimentFunction],
+        training_only_experiments: tp.Optional[tp.List[base.ExperimentFunction]] = None,
+        pareto_size: int = 12,
+        pareto_subset_methods: tp.List[str] = None,
+    ) -> tp.List["SpecialEvaluationExperiment"]:
+        """Returns a list of MultiExperiment, corresponding to MOO cross-validation:
+        Each experiments consist in optimizing all but one of the input ExperimentFunction's,
+        and then considering that the score is the performance of the best solution in the
+        approximate Pareto front for the excluded ExperimentFunction.
+
+        Parameters
+        ----------
+        experiments: sequence of ExperimentFunction
+            iterable of experiment functions, used for creating the crossvalidation.
+        training_only_experiments: sequence of ExperimentFunction
+            iterable of experiment functions, used only as training functions in the crossvalidation and never for test..
+        pareto_size: int
+            if provided, selects a subset of the full pareto front with the given maximum size
+        subset: str
+            method for selecting the subset (see optimizer.pareto_front)
+
+        """
+        funcs: tp.List["SpecialEvaluationExperiment"] = []
+        if pareto_subset_methods is None:
+            pareto_subset_methods = ["random", "loss-covering", "EPS", "domain-covering", "hypervolume"]
```

Comment:
if we are bound to test all these as defaults, maybe we'll need to be able to return several outputs for one run (one for each subset). I feel doing different experiments altogether may be a waste of resources

Reply:
True.... our structure is not adapted to this.

Maybe this is something which should be moved to scripts.

I love the idea of a scripts directory for adding examples -- that would help users and disconnect applications from the core.